packages/itwin/tree-widget/src/components/trees/models-tree/ModelsTree.tsx

```diff
@@ -68,18 +73,24 @@ export interface ModelsTreeProps extends BaseFilterableTreeProps {
    * Custom visibility handler.
    */
   modelsVisibilityHandler?: ModelsVisibilityHandler | ((props: ModelsVisibilityHandlerProps) => ModelsVisibilityHandler);
+  /**
+   * Flag that determines if hierarchy level filtering will be enabled for this tree.
+   * @beta
+   */
+  isHierarchyFilteringEnabled?: boolean;
```

Comment:
Any reason it wasn't added to other tree components?

Reply:
Me and Saulius agreed to only do ModelsTree for now. My thinking was that it would be nice to have separate PR's (at least for the first one) since it makes reviewing PR's easier and after this completes I will have a better idea on how to implement this feature on other tree components, which will save time in my opinion.